action.yaml

```
@@ -0,0 +1,57 @@
+name: Bootstrap
+description: Bootstrap workflow setup for organization repositories
+author: Incognia
+inputs:
+  mask-variables:
+    description: Environment variables to mask (GITHUB_APP_PK and DOCKER_AUTHS already included)
+    default: ''
+  clone-submodules:
+    description: Include submodules in checkout
+    default: recursive
+runs:
+  using: composite
+  steps:
+  - name: Mask secret variables
+    shell: bash
+    run: |
+      echo "::add-mask::GITHUB_APP_PK"
+      echo "::add-mask::DOCKER_AUTHS"
+      for VAR in ${{ inputs.mask-variables }}
+      do
+        echo "::add-mask::${!VAR}"
+      done
+
+  - name: Import host environment variables
+    shell: bash
+    run: |
+      awk 'BEGIN{for(v in ENVIRON) print v}' | while read e; do
+        echo "$e<<EOF" >> $GITHUB_ENV
+        echo "${!e}" >> $GITHUB_ENV
+        echo 'EOF' >> $GITHUB_ENV
+      done
+
+  - name: Clone repository with regular authentication
+    uses: actions/checkout@v2
+    if: |
+      !env.GITHUB_APP_ID || !env.GITHUB_APP_PK
+    with:
+      submodules: ${{ inputs.clone-submodules }}
+      # needed so we have the reference of the latest tag in the tree
+      fetch-depth: '0'
+
+  - name: Generate access token
+    id: access-token
+    uses: inloco/actions-githubapp-installationaccesstoken@HEAD
+    if: env.GITHUB_APP_ID && env.GITHUB_APP_PK
+    with:
+      app-id: ${{ env.GITHUB_APP_ID }}
+      app-pk: ${{ env.GITHUB_APP_PK }}
+
+  - name: Clone repository with generated token for authentication
+    uses: actions/checkout@v2
+    if: env.GITHUB_APP_ID && env.GITHUB_APP_PK
+    with:
+      token: ${{ steps.access-token.outputs.instl-token }}
+      submodules: ${{ inputs.clone-submodules }}
+      # needed so we have the reference of the latest tag in the tree
+      fetch-depth: '0'
```

This can be simplified. Additionally, do we have a use-case where we don't want to clone submodules?

```suggestion
  - name: Installation Access Token
    id: iat
    uses: inloco/actions-githubapp-installationaccesstoken@HEAD
    if: env.GITHUB_APP_ID && env.GITHUB_APP_PK
    with:
      app-id: ${{ env.GITHUB_APP_ID }}
      app-pk: ${{ env.GITHUB_APP_PK }}
  - name: Checkout
    uses: actions/checkout@v2
    with:
      fetch-depth: 0
      submodules: recursive
      token: ${{ steps.iat.outputs.instl-token || github.token }}
```